src/main/java/de/feu/massim22/group3/agents/belief/Belief.java

```diff
@@ -1196,14 +1213,39 @@ public AgentReport getAgentReport() {
         }
         int distGoalZone = 999;
         Point nearestGoalZone = new Point(0, 0);
+        List<Point> uniqueGoalZones = new ArrayList<Point>();
+        Point gameMapSize = Navi.<INaviAgentV1>get().getGameMapSize(getAgentShortName());
         for (ReachableGoalZone goalZone : reachableGoalZones) {
             if (goalZone.distance() < distGoalZone) {
                 distGoalZone = goalZone.distance();
                 nearestGoalZone = goalZone.position();
             }
+            if (uniqueGoalZones.size() == 0) {
+                uniqueGoalZones.add(goalZone.position());
+            }
+            else {
+                for (Point uniqueGoalZone : uniqueGoalZones) {
+                    Point gz = DirectionUtil.normalizePointOntoMap(new Point(goalZone.position()), gameMapSize);
+                    Point unique_gz = DirectionUtil.normalizePointOntoMap(uniqueGoalZone, gameMapSize);
+                    if (Math.abs(gz.x - unique_gz.x) + Math.abs(gz.y - unique_gz.y) > 15) {
+                        uniqueGoalZones.add(goalZone.position());
```

Comment:
Funktioniert glaube ich mit mehr als 2 goal zones nicht mehr. Prinzipiell wüsste der Supervisor auch schon wie viele gefundene goalzones es gibt

Reply:
Die GoalZone-Erkennung im Supervisor habe ich nicht zuverlässig hinbekommen. Da kamen am Ende viel mehr GoalZones heraus als es wirklich gibt. Im Belief hat man hingegen zu jedem Zeitpunkt genau die vorhandenen Reachable goal zones gleichzeitig drin.. Deswegen habe ich es erstmal in den Belief gelegt und da ist es wirklich nur für 2 goal zones ausgelegt. Nach dem Tunier schaue ich es mir evtl. noch einmal an. Dieses ganze Message-senden macht das ganze schon irgendwie unhandlich.